### PR TITLE
add -namespace flag to manage only CRDs in that namespace

### PR DESCRIPTION
**Pull request by scotthew1**
change for #172

### CODE REVIEW

**Comment by scotthew1** [comment]
signed the CLA

File: main.go [hunk 1]
```diff
@@ -60,6 +60,8 @@ var (
 	submissionRunnerThreads = flag.Int("submission-threads", 3, "Number of worker threads "+
 		"used by the SparkApplication submission runner.")
 	resyncInterval = flag.Int("resync-interval", 30, "Informer resync interval in seconds")
+	namespace = flag.String("namespace", "", "The Kubernetes namespace to manage. "+
+		"Will manage CRDs for the whole cluster if unset.")
```
Comment (liyinan926) [inline main.go]: Changing `CRDs` to `custom resource objects of the managed CRD types` makes it clearer.

File: main.go [hunk 2]
```diff
@@ -105,12 +107,17 @@ func main() {
 		}
 	}
 
-	factory := crdinformers.NewSharedInformerFactory(
+	var factoryOpts []crdinformers.SharedInformerOption
+	if *namespace != "" {
```
Comment (liyinan926) [inline main.go]: I think we can be more defensive by doing `if namespace != nil && *namespace != ""`.

Comment (scotthew1) [reply main.go]: ah good call

File: main.go [hunk 3]
```diff
@@ -60,6 +60,8 @@ var (
 	submissionRunnerThreads = flag.Int("submission-threads", 3, "Number of worker threads "+
 		"used by the SparkApplication submission runner.")
 	resyncInterval = flag.Int("resync-interval", 30, "Informer resync interval in seconds")
+	namespace = flag.String("namespace", "", "The Kubernetes namespace to manage. "+
```
[the remark this answers is not in the document]
Comment (scotthew1) [reply main.go]: i like this idea 👍

File: main.go [hunk 4]
```diff
@@ -105,12 +108,20 @@ func main() {
 		}
 	}
 
-	factory := crdinformers.NewSharedInformerFactory(
+	var factoryOpts []crdinformers.SharedInformerOption
+	if namespace == nil {
```
[the remark this answers is not in the document]
Comment (scotthew1) [reply main.go]: ok, i thought this was the case before where the default value was an empty string, but wanted to keep my bases covered after your previous comment. i will clean this up.

**Pull request merged**